org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalDescriptionProvider.java

```diff
@@ -385,9 +405,11 @@ private void createOverrideMethodProposalLabel(CompletionProposal methodProposal
 		item.setInsertText(name);
 		nameBuffer.append(name);
 		// parameters
-		nameBuffer.append('(');
-		appendUnboundedParameterList(nameBuffer, methodProposal);
-		nameBuffer.append(')');
+		StringBuilder parameters = new StringBuilder();
+		parameters.append('(');
+		appendUnboundedParameterList(parameters, methodProposal);
+		parameters.append(')');
+		nameBuffer.append(parameters);
```

Comment:
A small suggestion about the code: we can lazy build the `nameBuffer`.

Pseudocode may look like this:

```
1. Get method name
2. Get method parameters
3. Get method return type

if (isCompletionItemLabelDetailsSupport()) {
    setLabelDetails(...)
} else {
    build the `nameBuffer` and set it to the item label
}
```

I think that can make the code easier to read.